README.md

```diff
@@ -28,13 +27,314 @@ pip install -U pyinstaller-hooks-contrib
+# The full license is available in LICENSE.GPL.txt, distributed with
+# this software.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ------------------------------------------------------------------
+```
+
+</details>
+
+<details><summary>APL header for runtime hooks only.
+Again, if you're unsure if your hook is a runtime hook then it'll be a standard hook.</summary>
+
+```python
+# ------------------------------------------------------------------
+# Copyright (c) 2021 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the Apache License 2.0
+#
+# The full license is available in LICENSE.APL.txt, distributed with
+# this software.
+#
+# SPDX-License-Identifier: Apache-2.0
+# ------------------------------------------------------------------
+```
+
+</details>
+
+
+If you are **updating** a hook, skip this step.
+Do not update the year of the copyright header - even if it's out of date.
+
+
+### Test
+
+Having tests is key to our continuous integration.
+With them we can automatically verify that your hook works on all platforms, all Python versions and new versions of libraries as they are released.
+Without them, we have no idea if the hook is broken until someone finds out the hard way.
+Please write tests!!!
+
+That said, some user interface libraries may be impossible to test without user interaction
+or a wrapper library for some web API may require credentials (and possibly a paid subscription) to test.
+In such cases, don't provide a test.
+Instead explain in the commit message why an automatic test is impractical then skip on to [the next step](#run-linter).
+
+
+#### Write tests(s)
+
+A test should be the least amount of code required to cause a breakage
+if you do not have the hook which you are contributing.
+For example if you are writing a hook for a library called `foo`
+which crashes immediately under PyInstaller on `import foo` then `import foo` is your test.
+If `import foo` works even without the hook then you will have to get a bit more creative.
+Good sources of such minimal tests are the hello world examples
+from the documentation of whichever library you're writing a hook for.
+Tests should not explicitly check for presence of data files or hidden modules directly -
+rather they should use parts of the library which are expected to use said data files or hidden modules.
+
+Tests currently all live in [src/_pyinstaller_hooks_contrib/tests/test_libraries.py](../master/src/_pyinstaller_hooks_contrib/tests/test_libraries.py).
+Navigate there and add something like the following, replacing all occurrences of `foo` with the real name of the library.
+(Note where you put it in that file doesn't matter.)
+
+```python
+@importorskip('foo')
+def test_foo(pyi_builder):
+    pyi_builder.test_source("""
+
+        # Your test here!
+        import foo
+
+        foo.something_fooey()
+
+    """)
+```
+
+If need be, you can specify version constraints using `@importorskip('foo', minversion="1.2.3")`
```

Comment:
I think the `minversion` in the `importorskip` no longer works since https://github.com/pyinstaller/pyinstaller/pull/6023. Should it mention `@requires` instead?

Reply:
Yes, you're right. I've been meaning to address that for ages.